lib/main.dart

```diff
@@ -17,121 +17,122 @@ import 'node/os.dart';
 import 'node/process.dart';
 
 void main(List<String> args) async {
-  try {
-    // sdk
-    var sdk = core.getInput('sdk');
-    if (sdk.isEmpty) {
-      sdk = 'stable';
-    }
-
-    // flavor
-    var flavor = core.getInput('flavor');
-    if (flavor.isEmpty) {
-      flavor = sdk == 'main' ? 'raw' : 'release';
-    } else if (flavor != 'raw' && flavor != 'release') {
-      _fail("Unrecognized build flavor '$flavor'.");
-      return;
-    }
-    final raw = flavor == 'raw';
-
-    // os
-    final os = getPlatform();
+  // sdk
+  var sdk = core.getInput('sdk');
+  if (sdk.isEmpty) {
+    sdk = 'stable';
+  }
+  // A `3.0` in a workflow file reaches us as a `3` here; promote any int value
+  // back to a double.
+  if (int.tryParse(sdk) != null && !sdk.contains('.')) {
+    // Convert a '3' to a '3.0'.
+    sdk = '$sdk.0';
+  }
 
-    // architecture
-    var architecture = core.getInput('architecture');
-    if (architecture.isEmpty) {
-      architecture = getArch();
-    }
+  // flavor
+  var flavor = core.getInput('flavor');
+  if (flavor.isEmpty) {
+    flavor = sdk == 'main' ? 'raw' : 'release';
+  } else if (flavor != 'raw' && flavor != 'release') {
+    _fail("Unrecognized build flavor '$flavor'.");
+    // return;
```

Comment:
_fail is Never so remove this debug line

Same for below

Reply:
will do